apps/nerves_hub_web_core/lib/nerves_hub_web_core/products.ex

```diff
@@ -229,6 +229,25 @@ defmodule NervesHubWebCore.Products do
     Product.changeset(product, %{})
   end
 
+  def parse_json_data(%{"header" => %{"uniqueId" => serial}} = data) do
```

Comment:
```suggestion
  def parse_jose_data(%{"header" => %{"uniqueId" => serial}} = data) do
```

I skimmed the PDF. It sound like the intention is that their file follow Javascript Object Signing and Encryption (JOSE) and in theory, this should work for anyone else who exports JOSE files. How about renaming this and the following references `jose`?